internal/controller/serving/reconcilers/model_registry_reconciler.go

```diff
@@ -29,12 +29,16 @@ func NewModelRegistryInferenceServiceReconciler(client client.Client, log logr.L
 	if len(dscList.Items) > 0 {
 		dsc = dscList.Items[0]
 
-		ns, found, err := unstructured.NestedFieldCopy(dsc.Object, "spec", "components", "modelRegistry", "registriesNamespace")
-		if err == nil && found {
+		ns, found, err := unstructured.NestedFieldCopy(dsc.Object, "spec", "components", "modelregistry", "registriesNamespace")
+		if err != nil || !found {
+			log.Error(err, "Failed to get Model Registry Namespace from DataScienceCluster")
```

Comment:
maybe log as a warning if it is not a blocker and inform the user what namespace will be set instead.

Reply:
wait, is there no default NS if it isn't found in the DSC?
What happens if there is no NS set? If that's the case we should return a error to the reconciler.

Reply:
I think there should not be a case when the field is empty, looking at the definition in the DSC type:

```go
	// Namespace for model registries to be installed, configurable only once when model registry is enabled, defaults to "odh-model-registries"
	// +kubebuilder:default="odh-model-registries"
	// +kubebuilder:validation:Pattern="^([a-z0-9]([-a-z0-9]*[a-z0-9])?)?$"
	// +kubebuilder:validation:MaxLength=63
	RegistriesNamespace string `json:"registriesNamespace,omitempty"`
```

but yeah I agree with you, it's better to add the possibility to return an error, maybe in the case of modelregistry not being in the "Managed" state

Reply:
oh, ok, if there is the default one being set it is fine, we just need to make sure at the code level that it is set.

Reply:
added a check in https://github.com/opendatahub-io/odh-model-controller/pull/369/commits/eb2017532b14bbb735d5f15b7a665d22332d1fb4